### ISSUE TITLE
About "Sigmoid"

### ISSUE THREAD

**Issue by likyoo** [GeoZcx/A-deeply-supervised-image-fusion-network-for-change-detection-in-remote-sensing-images#1]
In the pytorch version code:

1. 'Sigmoid' is used when you build DSIFN model, but it's used again in your loss function.

2. Then, in loss.py, why use 'Sigmoid' function for 'target' ?

**Comment by GeoZcx** [comment]
The motivation of using sigmoid on the output value can be referred to our paper: "if y_i gets zero, -log⁡(y_i ) will be infinitely great, an infinitely great loss will corrupt the network. In our method, we use sigmoid binary cross-entropy as part of our loss function. "

**Comment by likyoo** [comment]
Yes, thanks for your answer.
But, in BCELoss() of pytorch, "[Our solution is that BCELoss clamps its log function outputs to be greater than or equal to -100. This way, we can always have a finite loss value and a linear backward method.](https://pytorch.org/docs/stable/search.html?q=BCELoss&check_keywords=yes&area=default)". So there is no infinity here.
And if you use Sigmoid twice, I think the output will be biased. Could you provide some result of ablation experiment about this?

**Comment by GeoZcx** [comment]
Thanks for your reminder! Honestly, I did not check the detailed implementation of bceloss in pytorch, so I code the Pytorch version the same way as I did in the Keras version. It seems that  Pytorch sets a minimum value (i.g.,-100) to bceloss to avoid the infinity problem. So maybe discarding the sigmoid function in Pytorch is also applicable. I did not conduct the ablation experiment in the Pytorch version, but in the Keras version, the infinity problem does exist. 
As for the concern of 'bias' caused by using Sigmoid twice,  since the output and the target are both mapped into the same field, they can be matched up with each other. Let's say, if we have a target field->(0,1) and an output field->(0,1), after the sigmoid function, we get the sig(target) field->sig(0,1) and the sig(output) field->sig(0,1). The two fields align with each other. So maybe it can answer the 'bias'  you mentioned above?
